api/API.md

```diff
@@ -221,81 +220,30 @@ Response:
-
-The request mime type MUST be "application/x-proto-binary".
-The request MUST contain a message with list of certificates to use with this edge node. The body MUST be a protobuf message of type [certs.ZEdgeNodeCert](./proto/certs/certs.proto).
-
-Response:
-
-The response MUST NOT contain any body content.
-
-### Attestation Nonce
-
-Retrieve a nonce to use in Edge node while constructing the next attestation quote. The next attestation quote published by the edge node MUST include this nonce
-
-   POST /api/v1/edgeDevice/attestNonce
-
-Return codes:
-
-* Unauthenticated or invalid credentials: `401`
-* Valid credentials without authorization: `403`
 * Success: `201`
 * Unknown Device: `400`
 * Missing or unprocessable body: `422`
 
 Request:
 
-The request MUST use the Device certificate for mTLS authentication.
-
-The request mime type MUST be "application/x-proto-binary".
-The body MUST be a protobuf message of type [attest.ZAttestNonceReq](./proto/attest/attest.proto).
-
-Response:
-
-The request mime type MUST be "application/x-proto-binary".
-The body MUST be a protobuf message of type [attest.ZAttestNonceResp](./proto/attest/attest.proto).
-
-### Attestation Quote
-
-Publish PCR quote computed by Trusted Platform Module, by including the latest nonce provided by the Controller
-
-   POST /api/v1/edgeDevice/attestQuote
-
-Return codes:
-
-* Unauthenticated or invalid credentials: `401`
-* Success: `201`
-* Unknown Device: `400`
-* Missing or unprocessable body: `422`
-
-Request:
-
-The request MUST use the Device certificate for mTLS authentication.
-
 The request mime type MUST be "application/x-proto-binary".
-The body MUST be a protobuf message of type [attest.ZAttestQuoteReq](./proto/attest/attest.proto).
+The body MUST be a protobuf message of type [attest.ZAttestReq](./proto/attest/attest.proto).
+`ZAttestReq` has `reqType` which MUST be filled by EVE to identify sub-type of the request. If `reqType` is `ATTEST_REQ_CERT`, `certs` should be filled with certificates used by EVE. If `isIMutable` attribute is not set, Controller should reject any request that would overwrite an existing certificate previously published. If `reqType` is `ATTEST_REQ_NONCE`, no other field needs to be set in `ZAttestReq`. If `reqType` is `ATTEST_REQ_QUOTE`, then `quote` MUST be populated with the quote generated by Trusted Platform Module.
 
 Response:
 
 The request mime type MUST be "application/x-proto-binary".
-The body MUST be a protobuf message of type [attest.ZAttestQuoteResp](./proto/attest/attest.proto).
+The body MUST be a protobuf message of type [attest.ZAttestResp](./proto/attest/attest.proto).
+`ZAttestResp` has `respType` which MUST be filled by Controller. If `respType` is `ATTEST_RESP_CERT`, no other field needs to be set. If `respType` is `ATTEST_RESP_NONCE`, nonce field MUST be set. If `respType` is `ATTEST_RESP_QUOTE`, then `quoteResp` MUST be filled with the result of the attestation.
```

Comment:
If the device sends a ATTEST_REQ_CERT with malformed PEM, how will you respond? Could use the http code for errors. But for ATTEST_REQ_QUOTE there is an explicit quoteResp for the success/fail.
So are we splitting format errors into http response code, and higher-level semantic issues (such as pass/fail) into the response payoad. Just trying to understand the approach

Reply:
Here the idea is: if you post a wrong/malformed cert,  controller should just silently discard that, as it would, when dealing with a corrupted payload(and return 400 Bad Request?). But quote, even if it is a legal quote, may not match the quote expected by cloud, so attestation result can be pass or fail.